client/src/test/scala/org/http4s/client/middleware/FollowRedirectSpec.scala

```diff
@@ -132,7 +133,7 @@ class FollowRedirectSpec extends Http4sSpec with Tables {
       val statefulService = HttpService[IO] {
         case GET -> Root / "loop" =>
           val body = loopCounter.incrementAndGet.toString
-          MovedPermanently(uri("/loop")).withBody(body)
+          MovedPermanently(uri("/loop")).flatMap(_.withBody(body))
```

Comment:
This is an example of a response generator that has a required header.  It should probably take the body and perhaps more variadic headers.  Then it would look like:

```scala
MovedPermanently(uri("/loop"), body)
```